Added workspace_symbol_picker

EbbDrop opened the pull request:
closes #1009
Uses `space` > `t` to open a workspace wide symbol picker.
![helix_wp](https://user-images.githubusercontent.com/5936614/140943452-03713c92-1b8d-42c6-a185-f3d7f3f36c32.png)

sudormrfbin commented:
The start of an item gets truncated if it's too long, which hides the name of symbol (due to #951):

![Screenshot_2021-11-09_20-52-45](https://user-images.githubusercontent.com/23398472/140952735-217f8cfb-3a52-46ce-be28-a743769fd46b.png)

EbbDrop commented:
I see two potential solutions:

1) Move the truncation to the other side for this picker but than we get the same problem as in #945:
```
Cell (src/m...
Cell (src/a...
```
2) Switch the file and symbol names:
```
...ain.rs: Cell
...elp.rs: Cell
```
But this does not look so good without truncation:
```
src/lib.rs: Cell
src/bin/main.rs: Cell
```

What solution would be preferred? Or does anybody have any other ideas.

archseer commented:
Let's add a field `pub truncate_start: bool` to `Picker`. It should default to `false`, and leave it out of `Picker::new` parameters, then use it here: https://github.com/helix-editor/helix/blob/cf831b1a65625f29d6e1bc12483a45c1adc8dff4/helix-term/src/ui/picker.rs#L569

This way you can initialize it like this:

```rust
let picker = Picker::new(...);
picker.truncate_start = true;
compositor.push(Box::new(picker))
```

By the way, what theme are you using? It looks like it's missing the picker border color, that's why it's blue.

archseer commented:
We probably want to set this flag on the other symbol picker too. I'm also wondering whether the key should be `space+S` to match the other picker maybe?

EbbDrop commented:
I and the other symbol picker uses FilePicker at the moment. Should i also add the truncate_start field to FilePicker?
ps.
I am using the gruvbox theme edited to have a transparent background but the blue is still there without my edits.

archseer commented:
You can keep `FilePicker` as is since we always want to show the end of the file path there.

EbbDrop commented:
FilePicker is used at the moment for all the pickers since it also gives the file preview to the left. So I think it is FilePicker that should get the change to have a truncate_left option.

archseer commented:
Ah sorry, I confused it with the `file_picker` method. Yeah, you'll need to add this to both `FilePicker` and `Picker`

EbbDrop commented:
Moved the truncation to the other side for both pickers and changed the shortcut to `space`+`S`

![Screenshot_2021-11-11_16-39-51](https://user-images.githubusercontent.com/5936614/141328308-2ada49a1-02b1-419c-b9ba-e891685546ed.png)

sudormrfbin commented on helix-term/src/commands.rs:
```diff
@@ -2748,6 +2749,69 @@ fn symbol_picker(cx: &mut Context) {
+                    },
+                    move |editor: &mut Editor, symbol, action| {
+                        let path = symbol.location.uri.to_file_path().unwrap();
+                        editor.open(path, action).expect("editor.open faild");
```
Missed an `e` here in `failed`.

sudormrfbin reviewed (approved):
LGTM, tested with Rust. There's also a merge conflict btw.

EbbDrop commented:
Fixed typo and merge conflict

archseer commented:
Thanks! 🎉
